src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
 
+        if (table[hash] == null) {
+            table[hash] = newNode;
+        } else {
+            Node<K, V> current = table[hash];
+            while (current != null) {
+                if (current.hash == hash
+                        && (current.key == key || (key != null && key.equals(current.key)))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        Node<K, V> node = table[hash];
+
+        while (node != null) {
+            if (node.hash == hash
+                    && (node.key == key || (key != null && key.equals(node.key)))) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % DEFAULT_CAPACITY);
```

Comment:
This `hash` method might not work correctly when you resize your `table`. It uses `DEFAULT_CAPACITY` for calculating the hash of the key, but after resizing, the capacity of your `table` will change, and you should use the current capacity for calculating the hash. You can move this logic to the `put` method or pass current `table` capacity as a parameter to the `hash` method.

Reply:
but in the resize() method I override the hashcode.
int newHash = Math.abs(node.hash % newCapacity);